yt_dlp/utils/_utils.py

```diff
@@ -2099,7 +2099,9 @@ def prepend_extension(filename, ext, expected_real_ext=None):
 
 def replace_extension(filename, ext, expected_real_ext=None):
     name, real_ext = os.path.splitext(filename)
-    return '{}.{}'.format(
+    ext = ext if ext.startswith('.') else '.' + ext
+
+    return '{}{}'.format(
```

Comment:
For viz. I've updated this method to allow you to pass an `ext` with or without a leading `.`.

This doesn't impact existing usage and makes it easier to pipe together existing commands like:

```python
_, ext = os.path.splitext('foo/bar/file.mp4')
replace_extension(filename, ext)
```